unionml/model.py

```diff
@@ -109,6 +109,8 @@ class in sklearn, ``Module`` in pytorch, etc.
         self._train_task = None
         self._predict_task = None
         self._predict_from_features_task = None
+        self.__callbacks: Dict[str, Callable] = {}
+        self.__callback_tasks: Dict[str, Any] = {}
```

Comment:
callbacks-as-tasks won't be necessary if we invoke them inside the `@inner_task(...)(predict_task)` function, as suggested above.